src/KafkaW/Consumer.cpp

```diff
@@ -129,13 +130,29 @@ bool Consumer::topicPresent(const std::string &TopicName) {
 
 void Consumer::updateMetadata() {
   RdKafka::Metadata *MetadataPtr = nullptr;
-  auto ErrorCode = KafkaConsumer->metadata(
-      true, nullptr, &MetadataPtr, ConsumerBrokerSettings.MetadataTimeoutMS);
-  if (ErrorCode == RdKafka::ERR__TRANSPORT)
-    throw MetadataException("Cannot contact broker");
-  else if (ErrorCode != RdKafka::ERR_NO_ERROR) {
-    throw MetadataException(
-        "Consumer::updateMetadata() - error while retrieving metadata.");
+  bool Connected = false;
+  short LoopCounter = 0;
+  while (!Connected) {
+    auto ErrorCode = KafkaConsumer->metadata(
```

Comment:
The logic for the metadata query itself (basically what the method did before your changes) could be refactored to a separate method, which returns the `Connected` bool. I think that would improve readability as it would separate the query logic from the retry logic.